homeassistant/components/automation/__init__.py

```diff
@@ -31,17 +33,77 @@
 
 DEFAULT_CONDITION_TYPE = CONDITION_TYPE_AND
 
+METHOD_TRIGGER = 'trigger'
+METHOD_IF_ACTION = 'if_action'
+
 _LOGGER = logging.getLogger(__name__)
 
 
+def _ensure_list(value):
+    """Wrap value in list if it is not one."""
+    return value if isinstance(value, list) else [value]
```

Comment:
What do you expect here? Iterating over a dictionary returns the keys, so the list consists of all the keys in the dictionary.
Ahh, this is for the cases where someone can either specify a single value, or a list of values. Yes then the code you had does the right thing.